### ISSUE TITLE
Support kind and apiVersion as const

### ISSUE THREAD

**Issue by danielepolencic** [tommy351/kubernetes-models-ts#4]
The library knows the apiVersion and kind for each resource.

```ts
> console.log(new apps.v1.Deployment({}))
IoK8sApiAppsV1Deployment { apiVersion: 'apps/v1', kind: 'Deployment' }
```

However, the types are 'string | undefined'

![image](https://user-images.githubusercontent.com/1130499/56814596-1e92ac00-6872-11e9-9f44-17269cb9fcd5.png)

It would be convenient to have the type and kind `as const` so that I can have proper narrowing in this switch statement:

```ts
type Resource = v1.Pod | apps.v1.Deployment
function hello(resource: Resource) {
  switch (resource.kind) {
    case v1.Pod.kind: {
      /* resource is a Pod */
    }
    case apps.v1.Deployment.kind: {
      /* resource is a Deployment */
    }
    default:
      throw new Error('Oops') // resource is never
  }
}
```

**Comment by tommy351** [comment]
This feature is implemented in v0.6.0.

**Comment by danielepolencic** [comment]
Great news! Thanks for this 🙇
